src/renderers/dom/fiber/ReactDOMFiber.js

```diff
@@ -347,34 +354,49 @@ var DOMRenderer = ReactFiberReconciler({
     parentInstance: Instance | Container,
     child: Instance | TextInstance,
   ): void {
-    parentInstance.appendChild(child);
+    if (parentInstance.nodeType === COMMENT_NODE) {
```

Comment:
This is the check I was worried about. I wonder if we should have a different set of helpers that only apply when `parentInstance` is Container so that this only needs to apply on the top level.

This check probably isn't *that* slow but it is for every insertion and we want that to be __super__ fast if we for example go through this path for optimized paths that creates DOM nodes from server rendered markup.